File: Data Structures & Algorithms/maximum-product-subarray/submission-1.java
class Solution {
    public int maxProduct(int[] nums) {
        int n = nums.length;
        int currMin = 1;
        int currMax = 1;

        int res = nums[0];

        for(int num: nums) {
            int tmp1 = num * currMax;
            int tmp2 = num * currMin;

            currMax = Math.max(Math.max(tmp1, tmp2), num);
            currMin = Math.min(Math.min(tmp1, tmp2), num);

            res = Math.max(currMax, res);
        }

        return res;
    }
}

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.py
from heapq import heappush, heappop
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        pq = []

        for num in nums:
            heappush(pq, num)

            while len(pq) > k:
                heappop(pq)

        return pq[0]